maestro-cli/src/main/java/maestro/cli/update/Updates.kt

```diff
@@ -0,0 +1,102 @@
+package maestro.cli.update
+
+import maestro.cli.api.ApiClient
+import maestro.cli.api.CliVersion
+import maestro.cli.view.red
+import java.nio.file.Paths
+import java.util.Properties
+import java.util.UUID
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import kotlin.io.path.createDirectories
+import kotlin.io.path.exists
+import kotlin.io.path.readText
+import kotlin.io.path.writeText
+
+object Updates {
+
+    private val BASE_API_URL = "https://api.mobile.dev"
+    private val OS_NAME = System.getProperty("os.name")
+    private val FRESH_INSTALL: Boolean
+    private val DEVICE_UUID: String
+    private val CLI_VERSION: CliVersion? = getVersion().apply {
+        if (this == null) {
+            System.err.println("\nWarning: Failed to parse current version".red())
+        }
+    }
+
+    private val DEFAULT_THREAD_FACTORY = Executors.defaultThreadFactory()
+    private val EXECUTOR = Executors.newCachedThreadPool {
+        DEFAULT_THREAD_FACTORY.newThread(it).apply { isDaemon = true }
+    }
+
+    private var future: CompletableFuture<CliVersion?>? = null
+
+    init {
+        val uuidPath = Paths.get(System.getProperty("user.home"), ".maestro", "uuid")
+        FRESH_INSTALL = if (uuidPath.exists()) {
+            false
+        } else {
+            uuidPath.parent.createDirectories()
+            uuidPath.writeText(UUID.randomUUID().toString())
+            true
+        }
+        DEVICE_UUID = uuidPath.readText()
+    }
+
+    fun fetchUpdatesAsync() {
```

Comment:
Logic seems correct but just to double check - none of those operations print an error if you are not connected to the Internet?

Reply:
Yep that's right